samples/ProjectGrandmaster.Unity/Assets/Script/Board/BoardInformation.cs

```diff
@@ -0,0 +1,633 @@
+
+        // Side that won. -1 if black, 0 if draw, +1 if white.
+        // Game Ended must also be true
+        public int Winner { get; set; }
+
+        // Arrow on the boardmenu that let's the player know if its their turn
+        // Green if true, else red. 
+        public GameObject whiteArrow;
+        public GameObject blackArrow;
+        public Material playerTurn;
+        public Material opponentTurn;
+
+        // Variables related to pawn promotion
+        public Mesh mesh { get; set; }
+        public bool promoted { get; set; }
+        public bool meshChosen { get; set; }
+        public GameObject pawnPromo;
+
+        public GameObject boardMenuTileText;
+        public GameObject boardMenuTileTextTwo;
+        public Mesh pawnMesh;
+
+        void Start()
+        {
+            /// <summary>
+            /// 2D chess layout with individual piece GameObject
+            /// </summary>
+            board = new GameObject[8, 8];
+            piecesOnBoard = new List<GameObject>();
+
+            foreach (GameObject piece in GameObject.FindGameObjectsWithTag("pieces"))
+            {
+                PieceInformation info = piece.GetComponent<PieceInformation>();
+                int x = info.GetXPosition();
+                int z = info.GetZPosition();
+                board[z, x] = piece;
+
+                /// <summary>
+                /// Initialise kings
+                /// </summary>
+                if (info.type == PieceInformation.Type.King)
+                {
+                    if (info.colour == PieceInformation.Colour.White) { whiteKing = piece; }
+                    else { blackKing = piece; }
+                }
+
+                /// <summary>
+                /// Add piece to piecesOnBoard list to keep track of all pieces on board
+                /// </summary>
+                piecesOnBoard.Add(piece);
+            }
+
+            /// <summary>
+            /// Assign null for empty positions
+            /// </summary>
+            for (int i = 2; i <= 5; i++)
+            {
+                for (int j = 0; j <= 7; j++)
+                {
+                    board[i, j] = null;
+                }
+            }
+            /// <summary>
+            /// Initialise whose turn it is
+            /// </summary>
+            turn = Colours.White;
+
+            /// <summary>
+            /// Initialise Data Structure
+            /// </summary>
+            pieceAction = GetComponent<PieceAction>();
+
+            //Turn off collisions for all pieces if ghost animation is active
+            if (ghostActive)
+            {
+                Physics.IgnoreLayerCollision(14, 14);
+                Physics.IgnoreLayerCollision(15, 15);
+                Physics.IgnoreLayerCollision(14, 15);
+            }
+
+            // Disable the result display
+            EndGameResult.SetActive(false);
+            pawnPromo.SetActive(false);
+        }
+
+        public GameObject[,] GetBoard() { return board; }
```

Comment:
In C# convention, we would make these properties:

public GameObject[,] Board { get { return board; } }

Reply:
Also, put them just below the fields.

Reply:
Changed field to
`public GameObject[,] Board { get; } = new GameObject[8, 8];`